src/driver/op.rs

```diff
@@ -166,16 +218,30 @@ impl Lifecycle {
 
         match mem::replace(self, Lifecycle::Submitted) {
             Lifecycle::Submitted => {
-                *self = Lifecycle::Completed(result, flags);
+                *self = Lifecycle::Completed(smallvec!((result, flags)));
                 false
             }
             Lifecycle::Waiting(waker) => {
-                *self = Lifecycle::Completed(result, flags);
+                *self = Lifecycle::Completed(smallvec!((result, flags)));
                 waker.wake();
                 false
             }
-            Lifecycle::Ignored(..) => true,
-            Lifecycle::Completed(..) => unreachable!("invalid operation state"),
+            Lifecycle::Ignored(op) => {
+                if io_uring::cqueue::more(flags){
+                    *self = Lifecycle::Ignored(op);
+                    false
+                } else {
+                    true
+                }
+            },
+            Lifecycle::Completed(mut v) => {
+                // We've run a data race between a multi-completion event and
+                // the first call to poll().
+                v.push((result,flags));
+                *self = Lifecycle::Completed(v);
+                false
+            }
```

Comment:
I believe the growth of the smallvec is only bound by the number of CQE's which can occur without acknowledgement
this is one of:
  - <= 64 (for UDP send_zc, with 64 GSO segments)
  - ? for tcp
  - the number of pre-registered buffers, assuming something like multishot recv is also implemented here
  
I think the idea would be able to force the future to consume the results, but this is problematic for 2 reasons
  - we have no info on the op here
  - it would pose a problem should a stream be implemented atop this, as you are likely to want to emit results based on partial updates
  
It might also possible perform op poll during creating, registering the waker prior to submission, eliminating the Submitted state, and adding a waker state to Completed.